### PR TITLE
fix #19304 Borrowing std/times.format causes Error: illformed AST

### LINKED ISSUE #19304: Borrowing std/times.format causes "Error: illformed AST"
Hello everybody,

For some stuff that I'm doing with norm I need a type that is, effectively, `DateTime`, but not in the eyes of the type system.
As such, I wanted to create a distinc ttype of `DateTime` and borrow the functionality of it that I needed. 
However, I receive an error when trying to borrow `times.format`

### Example
```nim
## contained in utils/djangoDateTime/djangoDateTimeType
import times

type DjangoDateTime* = distinct DateTime

proc toTime*(x: DjangoDateTime): Time  {.borrow.} # <-- works
proc format*(x: DjangoDateTime, f: string, loc: DateTimeLocale = DefaultLocale): string {.borrow.} # <-- throws Error: illformed AST
```

### Current Output
I couldn't find an issue about format and illformed AST, so I assume that's alright.
```
/home/isofruit/dev/nim/utils/djangoDateTime/djangoDateTimeType.nim(10, 6) Error: illformed AST: 
The terminal process "bash '-c', 'nim c -r -d:normDebug main.nim'" terminated with exit code: 1.
```

### Expected Output
- Error free compilation

### Possible Solution
I have no idea. I've got a workaround for those encountering the same bug though. Just make a custom format proc that converts to `DateTime` and calls its format proc:
```nim
proc format*(x: DjangoDateTime, f: string, loc: DateTimeLocale = DefaultLocale): string =
    let trueDt = x.DateTime 
    result = trueDt.format(f, loc)
```

```
$ nim -v
Nim Compiler Version 1.6.2 [Linux: amd64]
Compiled at 2021-12-17
Copyright (c) 2006-2021 by Andreas Rumpf

git hash: 9084d9bc02bcd983b81a4c76a05f27b9ce2707dd
active boot switches: -d:release
```

### PR DESCRIPTION
**Pull request by bung87**
fix #19304

### CODE REVIEW

File: compiler/ast.nim [hunk 1]
```diff
@@ -1430,6 +1430,12 @@ proc newSymNode*(sym: PSym, info: TLineInfo): PNode =
   result.typ = sym.typ
   result.info = info
 
+proc newSymNode*(sym: PSym, info: TLineInfo, typ: PType): PNode =
+  result = newNode(nkSym)
+  result.sym = sym
+  result.typ = typ
+  result.info = info
```
Comment (Araq) [inline compiler/ast.nim]: So now you introduced the notion of a nkSym node where `n.typ != n.sym.typ`. Unworkable.

Comment (bung87) [reply compiler/ast.nim]: so assign sym.typ to typ? here the typ is distinct base type or the param.typ, doesn't make much differience here.

Comment (metagn) [reply compiler/ast.nim]: Maybe use `nkConv`?

Comment (bung87) [reply compiler/ast.nim]: not sure, previous arguments are passed as `nkEmpty` node, but `nkEmpty` is used for `errorNode`, maybe make up fails cases show the problem.

**Comment by bung87** [comment]
@Araq merge this pls

**Pull request merged**